src/plugins/tts/utils.js

```diff
@@ -24,43 +24,61 @@ export function isAndroid(userAgent = navigator.userAgent) {
 const specialLangs =  {
   "zh-hans": "中文 (Zhōngwén)",
 };
+
 /**
  * @typedef {string} ISO6391
  * Language code in ISO 639-1 format. e.g. en, fr, zh
- * Can also retrieve language native name + ISO 639-1 code
  **/
 
 /**
  * @param {string} language in some format
- * @param {boolean?} getName gets Native Name + language code if true
  * @return {ISO6391?}
  */
-export function toISO6391(language, getName = false) {
+export function toISO6391(language) {
   if (!language) return null;
-  language = language.toLowerCase();
+  language = language.toLocaleLowerCase();
+  if (specialLangs[language]) {
+    return language;
+  }
+  const codeObj = findLanguage(language, 'iso639_1') || findLanguage(language, 'iso639_2T') || findLanguage(language, 'iso639_2B');
+  if (codeObj) return codeObj.iso639_1;
+  return null;
+}
 
-  return searchForISO6391(language, ['iso639_1'], getName) ||
-    searchForISO6391(language, ['iso639_2T'], getName) ||
-    searchForISO6391(language, ['iso639_2B', 'nativeName', 'name'], getName);
+/**
+ *
+ * @param {string} language
+ * @returns {string}
+ */
+export function toNativeName(language) {
+  if (!language) return null;
+  language = language.toLocaleLowerCase();
+  if (specialLangs[language]) {
+    return specialLangs[language];
+  }
+  const codeObj = findLanguage(language, 'iso639_1') || findLanguage(language, 'iso639_2T') || findLanguage(language, 'iso639_2B');
+  if (codeObj?.nativeName) return codeObj.nativeName.split(", ")[0];
+  return null;
 }
 
+/** @typedef {import('iso-language-codes').Code} Code */
+
 /**
- * Searches for the given long in the given columns.
  * @param {string} language
- * @param {Array<keyof import('iso-language-codes').Code>} columnsToSearch
- * @param {boolean?} getName
- * @return {ISO6391?}
+ * @returns {Code | null}
  */
-function searchForISO6391(language, columnsToSearch, getName) {
+function findLanguage(language, codeType) {
+  if (!language) return null;
+  language = language.toLowerCase();
   for (const lang of langs) {
-    for (const colName of columnsToSearch) {
-      if (lang[colName].split(', ').map(x => x.toLowerCase()).indexOf(language) != -1) {
-        if (getName) return `${lang.nativeName.split(", ")[0]} (${lang.iso639_1})`;
-        return lang.iso639_1;
-      }
+    if (lang[codeType].toLowerCase().split(", ").includes(language)) {
+      return lang;
+    } else if (lang.name.toLowerCase().split(", ").includes(language)) {
+      return lang;
+    } else if (lang.nativeName.toLowerCase().split(", ").includes(language)) {
+      return lang;
```

Comment:
This makes the signature of this method a little confusing ; it's exposing an interface like `findLanguage(query, field)`, but then secretly also checking the `name` and `nativeName` field. I think something similar to what was there before, where we iterate overall the requested field names, might be a clearer interface.